### PR TITLE
UAR-271: Update BO statements page (S2)

### PR DESCRIPTION
**Pull request by joshuahigginsonCH**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-271

### Change description
Added view, controller, unit tests and model changes.

### Work checklist

- [X] Tests added where applicable
- [X] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/model/update.type.model.ts [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
 
-export const UpdateKeys: string[] = [ "date_of_creation", "bo_mo_data" ];
+export const UpdateKeys: string[] = [ "date_of_creation", "bo_mo_data", "registrable_beneficial_owner" ];
 
 export interface Update {
-    date_of_creation?: string,
-    bo_mo_data?: true
+    date_of_creation?: string;
+    bo_mo_data?: true;
+    registrable_beneficial_owner?: string;
```
Comment (AndreKramer7654) [inline src/model/update.type.model.ts:9]: Is this not a boolean? has_new_or_ceased_registrable_beneficial_owner?

Comment (AndreKramer7654) [reply src/model/update.type.model.ts:9]: or an enum of two values? Which is harder to save out to the API.

File: src/model/update.type.model.ts [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
+}
+
+export enum RegistrableBeneficialOwnerType {
```
Comment (AndreKramer7654) [inline src/model/update.type.model.ts:12]: don't see this used in the post method

File: src/routes/index.ts [hunk 3]
```diff
@@ -487,4 +488,13 @@ router.route(config.UPDATE_FILING_DATE_URL)
...
   .post(updateFilingDate.post);
 
+router.route(config.UPDATE_REGISTRABLE_BENEFICIAL_OWNER_URL)
+  .all(
+    authentication,
+    companyAuthentication,
+    navigation.hasOverseasEntity
+  )
+  .get(updateRegistrableBeneficialOwner.get)
+  .post(...validator.registrableBeneficialOwner, checkValidations, updateRegistrableBeneficialOwner.post);
```
Comment (AndreKramer7654) [inline src/routes/index.ts:498]: Do you need validation to make sure one of the buttons is selected?

File: views/update/update-registrable-beneficial-owner.html [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+{% endblock %}
+
+{% block content %}
+<div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+        {% include "includes/list/errors.html" %}
+        <span class="govuk-caption-xl govuk-!-padding-bottom-1">{{ appData.entity_name }} - {{ appData.entity_number }}</span>
+        <form method="post">
+            {{ govukRadios({
+            errorMessage: errors.registrable_beneficial_owner if errors,
```
Comment (AndreKramer7654) [inline views/update/update-registrable-beneficial-owner.html:20]: Indentation

Comment (joshuahigginsonCH) [reply views/update/update-registrable-beneficial-owner.html]: Fixed

**Review by AndreKramer7654 — APPROVED** [review]

File: src/validation/error.messages.ts [hunk 5]
```diff
@@ -172,6 +172,8 @@ export enum ErrorMessages {
   SELECT_IF_SIGN_OUT = "Select yes if you are sure you want to sign out",
   SELECT_IF_CONTINUE_SAVED_APPLICATION = "Select yes if you want to continue with a saved application",
   SELECT_IF_ON_PUBLIC_REGISTER_IN_COUNTRY_FORMED_IN = "Select yes if it is already on a public register in the country it was formed in",
+  SELECT_IF_REGISTRABLE_BENEFICIAL_OWNER = "Select if anyone become or ceased to be a registrable beneficial owner during the update period",
```
Comment (bbell-companieshouse) [inline src/validation/error.messages.ts:175]: Should this be '..has became or ceased..' ?

File: test/__mocks__/text.mock.ts [hunk 6]
```diff
@@ -184,3 +184,4 @@ export const BENEFICIAL_OWNER_MANAGING_OFFICER_TYPE_LEGEND_TEXT = "Which type of
 export const UPDATE_BENEFICIAL_OWNER_TYPE_PAGE_REDIRECT = "Found. Redirecting to /update-an-overseas-entity/update-beneficial-owner-type";
 export const UPDATE_CHECK_YOUR_ANSWERS_CONTACT_DETAILS = "Contact details";
 export const UPDATE_CHECK_YOUR_ANSWERS_WHO_CAN_WE_CONTACT = "Who can we contact about this application?";
+export const UPDATE_REGISTRABLE_BENEFICIAL_OWNER_TITLE = "Has anyone become or ceased to be a registrable beneficial owner during the update period?";
```
Comment (bbell-companieshouse) [inline test/__mocks__/text.mock.ts:187]: Same comment as above

**Review by bbell-companieshouse — CHANGES_REQUESTED** [review]

File: test/__mocks__/text.mock.ts (hunk 6 above)
Comment (joshuahigginsonCH) [reply test/__mocks__/text.mock.ts:187]: This one is fine. Is just a check for the title

File: src/validation/error.messages.ts (hunk 5 above)
Comment (joshuahigginsonCH) [reply src/validation/error.messages.ts:175]: Addressed

**Review by bbell-companieshouse — APPROVED** [review]

**Pull request merged**